src/script/interpreter.cpp

```diff
@@ -1947,9 +2085,10 @@ std::vector<uint256> GetSpentScriptPubKeysSHA256(const std::vector<CTxOut>& outp
     std::vector<uint256> spent_spk_single_hashes;
     spent_spk_single_hashes.reserve(outputs_spent.size());
     for (const auto& txout : outputs_spent) {
-        CHashWriter ss(SER_GETHASH, 0);
-        ss << txout.scriptPubKey;
-        spent_spk_single_hashes.push_back(ss.GetSHA256());
+        // Normal serialization using the << operater would also serialize the lenght, therefore we directly write using CSHA256
+        uint256 spent_spk_single_hash;
+        CSHA256().Write(txout.scriptPubKey.data(), txout.scriptPubKey.size()).Finalize(spent_spk_single_hash.begin());
+        spent_spk_single_hashes.push_back(spent_spk_single_hash);
```

Comment:
For what it's worth, I think it was fine to serialize the length. Also spelling `length` in the comment.